### PR TITLE
Allow passing API key in individual calls

### PR DESCRIPTION
**Pull request by veechau**
Allow passing of API for individual calls in order to allow for thread safe usage of multiple API keys.

### CODE REVIEW

File: lib/checkr.rb [hunk 1]
```diff
@@ -61,6 +61,8 @@ def self.api_url(path='')
   end
 
   def self.request(method, path, params={}, headers={})
+    api_key = headers.key?(:api_key) ? headers.delete(:api_key) : @api_key
```
Comment (erdey) [inline lib/checkr.rb]: why do we need to delete the key?

Comment (jperichon) [reply lib/checkr.rb]: I'd recommend to set this as different param

**Comment by veechau** [comment]
Closing due to https://github.com/checkr/checkr-ruby/pull/53

File: lib/checkr.rb (hunk 1 above)
Comment (jperichon) [reply lib/checkr.rb]: Sorry I caused confusion with the usage of 'param' here. Ideally I wanted to see it in another argument. Given complexity for this, and that's it's been implemented like this (using headers) in other public gem, I'm ok with it.

**Pull request closed without merging**